pkg/mcs/resourcemanager/server/resource_group.go

```diff
@@ -252,13 +258,12 @@ func (rg *ResourceGroup) RequestRU(
 	sl *serviceLimiter,
 ) *rmpb.GrantedRUTokenBucket {
 	rg.Lock()
-	defer rg.Unlock()
-
 	if rg.RUSettings == nil || rg.RUSettings.RU.Settings == nil {
 		return nil
 	}
 	// First, try to get tokens from the resource group.
 	tb, trickleTimeMs := rg.RUSettings.RU.request(now, requiredToken, targetPeriodMs, clientUniqueID)
+	rg.Unlock()
```

Comment:
This is to fix the deadlock caused by line 273.

https://github.com/tikv/pd/pull/9577/files#diff-626c8564c95f18abf54adf61e8b99ec1457d48f587ee7b91d7969bfccceb2748R273